### ISSUE TITLE
JSON to be parsed somehow incorrect with numeric keys in result object.

### ISSUE THREAD

**Issue by BBigLee** [expressjs/body-parser#383]
In some cases, the parser can't parse request body correctly.
We expect the result to be like this:
![image](https://user-images.githubusercontent.com/13958569/68446282-44ccd780-0217-11ea-8cda-4bbdeacbb0c3.png)

But it turned out to be this:
![image](https://user-images.githubusercontent.com/13958569/68446369-7180ef00-0217-11ea-857e-05283510c897.png)

**Comment by dougwilson** [comment]
All this module does is use `JSON.parse`. Can you provide the following so we can reproduce the issue and diagnose what is happening?

1. Version of node.js this module and any other you are using.
2. Complete code that I can copy and paste and run without any modification to see the issue.
3. Complete instructions on how to run the above code and exactly how to replicate the issue.

Thank you.

**Comment by BBigLee** [comment]
> All this module does is use `JSON.parse`. Can you provide the following so we can reproduce the issue and diagnose what is happening?
> 
> 1. Version of node.js this module and any other you are using.
> 2. Complete code that I can copy and paste and run without any modification to see the issue.
> 3. Complete instructions on how to run the above code and exactly how to replicate the issue.
> 
> Thank you.

Thanks for your reply.
1. We use node 10.x
2. `const bodyParser = require('body-parser');
...

const app = express();
const session = cls.createNamespace('web-authentication');

app.set('trust proxy', true);

app.use((req, res, next) => {
  session.bindEmitter(req);
  session.bindEmitter(res);
  session.run(() => next());
});
app.use(bodyParser.json());
app.use(cookieParser());`

3. we can't replicate the issue, that why we want some help from you.

Thanks

**Comment by dougwilson** [comment]
There are so many variables, I'm not sure how I can help besides saying I have never seen anything like this so have no idea how to even guess at the issue. I would normally start with attaching a debugger to the process having the issue and set a conditional break point for when the condition occurs to dig through the state of the system.

**Comment by BBigLee** [comment]
Thanks @dougwilson 
I will try to find more useful info

**Comment by HarshithaKP** [comment]
@BBigLee Is this still an issue for you ?

**Comment by GrosSacASac** [comment]
Maybe you used the Object function which can give this kind of result
```
Object(JSON.stringify({email:"###PII-EXPOSURE"}))
```

String
​
0: "{"
​
1: "\""
​
2: "e"
​
3: "m"
​
4: "a"
​
5: "i"
​
6: "l"
​
7: "\""
​
8: ":"
​
9: "\""
​
10: "#"
​
11: "#"
​
12: "#"
​
13: "P"
​
14: "I"
​
15: "I"
​
16: "-"
​
17: "E"
​
18: "X"
​
19: "P"
​
20: "O"
​
21: "S"
​
22: "U"
​
23: "R"
​
24: "E"
​
25: "\""
​
26: "}"
​
length: 27
​
<prototype>: String { "" }
```

**Comment by BBigLee** [comment]
> @BBigLee Is this still an issue for you ?

Yes, but still can't find out how this would happen @HarshithaKP

**Comment by BBigLee** [comment]
> Maybe you used the Object function which can give this kind of result
> 
> ```
> Object(JSON.stringify({email:"###PII-EXPOSURE"}))
> ```
> 
> ```
> String
> ​
> 0: "{"
> ​
> 1: "\""
> ​
> 2: "e"
> ​
> 3: "m"
> ​
> 4: "a"
> ​
> 5: "i"
> ​
> 6: "l"
> ​
> 7: "\""
> ​
> 8: ":"
> ​
> 9: "\""
> ​
> 10: "#"
> ​
> 11: "#"
> ​
> 12: "#"
> ​
> 13: "P"
> ​
> 14: "I"
> ​
> 15: "I"
> ​
> 16: "-"
> ​
> 17: "E"
> ​
> 18: "X"
> ​
> 19: "P"
> ​
> 20: "O"
> ​
> 21: "S"
> ​
> 22: "U"
> ​
> 23: "R"
> ​
> 24: "E"
> ​
> 25: "\""
> ​
> 26: "}"
> ​
> length: 27
> ​
> <prototype>: String { "" }
> ```

@GrosSacASac I know this way will result to the issue situation, but I never used it in my code. And it failed only few times everyday, and I can't even reproduce manually, but found these errors in app logs.

**Comment by GrosSacASac** [comment]
OK, then close this issue until you can reproduce
